tests/tests_api/test_student.py

```diff
@@ -0,0 +1,91 @@
+from model_bakery import baker
+from rest_framework import status
+
+from api.models import AgeRange, DayAndTimeSlot, LanguageAndLevel, PersonalInfo, Student
+
+
+def test_student_create(api_client, faker):
+    initial_count = Student.objects.count()
+    personal_info = baker.make(PersonalInfo)
+    age_range_id = AgeRange.objects.first().id
+    teaching_languages_and_levels_ids = [
+        LanguageAndLevel.objects.first().id,
+        LanguageAndLevel.objects.last().id,
+    ]
+    availability_slots_ids = [
+        DayAndTimeSlot.objects.first().id,
+        DayAndTimeSlot.objects.last().id,
+    ]
+    m2m_fields = [
+        "teaching_languages_and_levels",
+        "availability_slots",
+    ]
+    data = {
+        "personal_info": personal_info.id,
+        "age_range": age_range_id,
+        "teaching_languages_and_levels": teaching_languages_and_levels_ids,
+        "is_member_of_speaking_club": faker.pybool(),
+        "requires_help_with_cv": faker.pybool(),
+        "availability_slots": availability_slots_ids,
+    }
+    response = api_client.post("/api/students/", data=data)
+
+    assert response.status_code == status.HTTP_201_CREATED
+    assert Student.objects.count() == initial_count + 1
+
+    # Changing for further filtering
+    for field in m2m_fields:
+        data[f"{field}__in"] = data.pop(field)
+
+    assert Student.objects.filter(**data).exists()
+
+
+def test_student_retrieve(api_client):
+    student = baker.make(Student, make_m2m=True)
+    # TODO: Student doesn't have `id` field, but `personal_info` does. Is it ok?
```

Comment:
`Student`, `Teacher`, `Coordinator` have one-on-one relationship to `PersonalInfo` and use `PersonalInfo` as primary key. I think that's fine?